src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -744,12 +815,59 @@ private boolean handleMaterials(@NotNull final Block block, @NotNull final IBloc
             return true;
         }
 
-        if (isBlockFree(block, block.getMetaFromState(blockState)))
+        List<ItemStack> itemList = new ArrayList<>();
+        itemList.add(BlockUtils.getItemStackFromBlockState(blockState));
+        if(job instanceof JobBuilder && ((JobBuilder) job).getStructure().getBlockInfo() != null && ((JobBuilder) job).getStructure().getBlockInfo().tileentityData != null)
         {
-            return true;
+            itemList.addAll(getItemStacksOfTileEntity(((JobBuilder) job).getStructure().getBlockInfo().tileentityData));
+        }
+
+        for(ItemStack stack: itemList)
+        {
+            if(!isBlockFree(stack) && checkOrRequestItems(stack))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     *((JobBuilder) job).getStructure().getBlockInfo().tileentityData
+     * @return
+     */
+    private List<ItemStack> getItemStacksOfTileEntity(NBTTagCompound compound)
+    {
+        List<ItemStack> items = new ArrayList<>();
+        TileEntity tileEntity = TileEntity.create(world, compound);
+        if(tileEntity instanceof TileEntityFlowerPot)
+        {
+            items.add(((TileEntityFlowerPot) tileEntity).getFlowerItemStack());
+        }
+        else if(tileEntity instanceof TileEntityLockable)
+        {
+            for(int i = 0; i < ((TileEntityLockable) tileEntity).getSizeInventory(); i++)
+            {
+                ItemStack stack = ((TileEntityLockable) tileEntity).getStackInSlot(i);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'stack' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)